feat: Added in tasks for op, sonieum, and ink sep (#830)
ethereum-optimism/superchain-ops: src/improvements/tasks/sep/005-opcm-upgrade-v300/VALIDATION.md

```

In mainnet runbooks, this calldata should appear in [Action Plan](https://gov.optimism.io/t/upgrade-proposal-14-isthmus-l1-contracts-mt-cannon/9796#p-43948-action-plan-9) section of the Governance proposal.

# State Changes

## Nested Safe State Overrides and Changes

This task is executed by the nested 2/2 `ProxyAdminOwner` Safe. Refer to the
[generic nested Safe execution validation document](../../../../../NESTED-VALIDATION.md)
for the expected state overrides and changes.

The `approvedHashes` mapping of the `ProxyAdminOwner` should change during the simulation.
See the ["Key Computation" section](../../../../../NESTED-VALIDATION.md#key-computation) in the nested validation doc
for instruction on how to validate this change.  You can find the target safe hash in the simulation output
under "Nested Multisig Child's Hash to Approve", which should be:

> - `0x7ae8800316ad25e257c77b54d034e5befc61dc38da97801125cc99f404be529e`

Reviewer: What is this hash? I don't see it references anywhere in my terminal or on Tenderly.

Reviewer: I removed this section.

Additionally, Safe-related nonces [will increment by one](../../../../../NESTED-VALIDATION.md#nonce-increments).

## Other Nonces
In addition to the Safe-related nonces mentioned [previously](#nested-safe-state-overrides-and-changes),
new contracts will also have a nonce value increment from 0 to 1.
This due to [EIP-161](https://eips.ethereum.org/EIPS/eip-161) which activated in 2016.

This affects the newly deployed dispute games mentioned in ["State Diffs"](#state-diffs).

## State Diffs

For each contract listed in the state diff, please verify that no contracts or state changes shown in the Tenderly diff are missing from this document. Additionally, please verify that for each contract:

- The following state changes (and none others) are made to that contract. This validates that no unexpected state
  changes occur.
- All addresses (in section headers and storage values) match the provided name, using the Etherscan and Superchain
  Registry links provided. This validates the bytecode deployed at the addresses contains the correct logic.
- All key values match the semantic meaning provided, which can be validated using the storage layout links provided.

> ### [`0x034edd2a225f7f429a63e0f1d2084b9e0a93b538` ](https://github.com/ethereum-optimism/superchain-registry/blob/1a5d7a208cea9b0ea175df1fe71bdc4da7f4c04c/superchain/configs/sepolia/op.toml#L5) (SystemConfig - OP Sepolia Testnet) Chain ID: 11155420

Reviewer: Can you remove all the chain names here? Having the chain Id is enough. We want a consistent experience across all VALIDATION files.

> ### [`0x034edd2a225f7f429a63e0f1d2084b9e0a93b538` ](https://github.com/ethereum-optimism/superchain-registry/blob/1a5d7a208cea9b0ea175df1fe71bdc4da7f4c04c/superchain/configs/sepolia/op.toml#L5) (SystemConfig - OP Sepolia Testnet) Chain ID: 11155420

Reviewer: ```suggestion
### [`0x034edd2a225f7f429a63e0f1d2084b9e0a93b538`](https://github.com/ethereum-optimism/superchain-registry/blob/1a5d7a208cea9b0ea175df1fe71bdc4da7f4c04c/superchain/configs/sepolia/op.toml#L5) (SystemConfig - OP Sepolia Testnet) Chain ID: 11155420
```

> ### [`0x034edd2a225f7f429a63e0f1d2084b9e0a93b538` ](https://github.com/ethereum-optimism/superchain-registry/blob/1a5d7a208cea9b0ea175df1fe71bdc4da7f4c04c/superchain/configs/sepolia/op.toml#L5) (SystemConfig - OP Sepolia Testnet) Chain ID: 11155420

Reviewer: ```suggestion
### [`0x034edd2a225f7f429a63e0f1d2084b9e0a93b538` ](https://github.com/ethereum-optimism/superchain-registry/blob/1a5d7a208cea9b0ea175df1fe71bdc4da7f4c04c/superchain/configs/sepolia/op.toml#L5) (SystemConfig) Chain ID: 11155420
```

> ### [`0x034edd2a225f7f429a63e0f1d2084b9e0a93b538` ](https://github.com/ethereum-optimism/superchain-registry/blob/1a5d7a208cea9b0ea175df1fe71bdc4da7f4c04c/superchain/configs/sepolia/op.toml#L5) (SystemConfig - OP Sepolia Testnet) Chain ID: 11155420

Reviewer: This link doesn't point to this address?

Author: Was missing a `9`, fixed now

- **Key:**          `0x360894a13ba1a3210667c828492db98dca3e2076cc3735a920a3ca505d382bbc`
  - **Decoded Kind:**      `address`
  - **Before:** `0x760C48C62A85045A6B69f07F4a9f22868659CbCc`

>   - **After:** [`0x340f923E5c7cbB2171146f64169EC9d5a9FfE647`](/Users/copypaste/Optimism/superchain-ops/src/improvements/tasks/sep/005-opcm-upgrade-v300/VALIDATION.md)

Reviewer: This isn't a proper link. Can you permalink to the actual address in the superchain registry.

Author: Contains the proper link now

  - **Summary:**           ERC-1967 implementation slot
  - **Detail:**            Standard slot for storing the implementation address in a proxy contract that follows the ERC-1967 standard.

  ---

### [`0x05c993e60179f28bf649a2bb5b00b5f4283bd525`](https://github.com/ethereum-optimism/superchain-registry/blob/00208555c3c356d6596feedb619da989de478ed7/superchain/configs/sepolia/ink.toml#L59-L59) (SystemConfig) - Chain ID: 763373

- **Key:**          `0x360894a13ba1a3210667c828492db98dca3e2076cc3735a920a3ca505d382bbc`
  - **Decoded Kind:**      `address`
  - **Before:** `0x760C48C62A85045A6B69f07F4a9f22868659CbCc`
  - **After:** `[0x340f923E5c7cbB2171146f64169EC9d5a9FfE647](https://github.com/ethereum-optimism/superchain-registry/blob/00208555c3c356d6596feedb619da989de478ed7/validation/standard/standard-versions-sepolia.toml#L9-L9)`
  - **Summary:**           ERC-1967 implementation slot

>   - **Detail:**            Standard slot for storing the implementation address in a proxy contract that follows the ERC-1967 standard.

Reviewer: ```suggestion
  - **Detail:**            Standard slot for storing the implementation address in a proxy contract that follows the ERC-1967 standard. SystemConfig contract for `op-contracts/v3.0.0-rc.2`.
```

  ---

### [`0x05f9613adb30026ffd634f38e5c4dfd30a197fa1`]((https://github.com/ethereum-optimism/superchain-registry/blob/00208555c3c356d6596feedb619da989de478ed7/superchain/configs/sepolia/op.toml#L63-L63)) (DisputeGameFactory) - Chain ID: 11155420

- **Key:**          `0x4d5a9bd2e41301728d41c8e705190becb4e74abe869f75bdb405b63716a35f9e`
  - **Before:**     `0x0000000000000000000000007717296cac5d39d362eb77a94c95483bebae214e`
  - **After:**     `0x000000000000000000000000845e5382d60ec16e538051e1876a985c5339cc62`

>   - **Summary:** Set a new game implementation for game type [PERMISSIONED_CANNON.](https://github.com/ethereum-optimism/optimism/blob/op-contracts/v2.0.0-rc.1/packages/contracts-bedrock/src/dispute/lib/Types.sol#L55")
>     - **Detail:** You can verify this slot corresponds to the game implementation for game type 1 by 
>                       deriving the slot value as follows:
>                       - Notice that [`gameImpls`](https://github.com/ethereum-optimism/optimism/blob/op-contracts/v2.0.0-rc.1/packages/contracts-bedrock/src/dispute/DisputeGameFactory.sol#L57") is a map from a `GameType` to a dispute game address.
>                       - Notice that `GameType` is [equivalent to a]("https://github.com/ethereum-optimism/optimism/blob/op-contracts/v2.0.0-rc.1/packages/contracts-bedrock/src/dispute/lib/LibUDT.sol#L224-L224").
>                       - Notice that the `gameImpls` [stored at slot 101]("https://github.com/ethereum-optimism/optimism/blob/op-contracts/v2.0.0-rc.1/packages/contracts-bedrock/snapshots/storageLayout/DisputeGameFactory.json#L41-L41"). 
>                       - Calculate the expected slot for game type 1 using `cast index &lt;KEY_TYPE&gt; &lt;KEY&gt; &lt;SLOT_NUMBER&gt;`:
>                         - `cast index uint32 1 101`
>                       - You should derive a value matching the "Raw Slot" here: 0x4d5a9bd2e41301728d41c8e705190becb4e74abe869f75bdb405b63716a35f   

Reviewer: ```suggestion
  - **Summary:**     Updates the implementation for game type 1.
  - **Detail:**    This is `gameImpls[1]` -> `0x6f67e57c143321e266bac32a0d9d22d88ce1b3e5`. The [`gameImpls` mapping](https://github.com/ethereum-optimism/optimism/blob/op-contracts/v3.0.0-rc.2/packages/contracts-bedrock/src/dispute/DisputeGameFactory.sol#L57) is at [storage slot 101](https://github.com/ethereum-optimism/optimism/blob/op-contracts/v3.0.0-rc.2/packages/contracts-bedrock/snapshots/storageLayout/DisputeGameFactory.json#L41) and is keyed by [`GameType` (`uint32`)](https://github.com/ethereum-optimism/optimism/blob/op-contracts/v3.0.0-rc.2/packages/contracts-bedrock/src/dispute/lib/LibUDT.sol#L224).
    - Confirm the expected key slot with the following:
      ```shell
      cast index uint32 1 101
      0x4d5a9bd2e41301728d41c8e705190becb4e74abe869f75bdb405b63716a35f9e
      ```
```

>   - **Summary:**  Set a new game implementation for game type [CANNON]("https://github.com/ethereum-optimism/optimism/blob/op-contracts/v2.0.0-rc.1/packages/contracts-bedrock/src/dispute/lib/Types.sol#L52")

Reviewer: Wheres the state diff for this? Shouldn't it include `0xffdfc1249c027f9191656349feb0761381bb32c9f557e01f419fd08754bf5a1b`?

Reviewer: I removed this section.

>   - **Summary:**  Set a new game implementation for game type [CANNON]("https://github.com/ethereum-optimism/optimism/blob/op-contracts/v2.0.0-rc.1/packages/contracts-bedrock/src/dispute/lib/Types.sol#L52")
>   - **Detail:** You can verify this slot corresponds to the game implementation for game type 0 by 
>                      deriving the slot value as follows:
>                      - Notice that [`gameImpls`]("https://github.com/ethereum-optimism/optimism/blob/op-contracts/v2.0.0-rc.1/packages/contracts-bedrock/src/dispute/DisputeGameFactory.sol#L57") is a map from a `GameType` to a dispute game address</a>.
>                      - Notice that `GameType` is [equivalent to a ]("https://github.com/ethereum-optimism/optimism/blob/op-contracts/v2.0.0-rc.1/packages/contracts-bedrock/src/dispute/lib/LibUDT.sol#L224-L224") `uint32`.                      
>                      - Notice that the `gameImpls` is [stored at slot 101]("https://github.com/ethereum-optimism/optimism/blob/op-contracts/v2.0.0-rc.1/packages/contracts-bedrock/snapshots/storageLayout/DisputeGameFactory.json#L41-L41"). 
>                      - Calculate the expected slot for game type 0 using `cast index &lt;KEY_TYPE&gt; &lt;KEY&gt; &lt;SLOT_NUMBER&gt;`:
>                        - `cast index uint32 0 101`
>                      - You should derive a value matching the "Raw Slot" here: 0xffdfc1249c027f9191656349feb0761381bb32c9f557e01f419fd08754bf5a1b 

Reviewer: ```suggestion
- **Key:**          `0xffdfc1249c027f9191656349feb0761381bb32c9f557e01f419fd08754bf5a1b`

  - **Before:**     `0x0000000000000000000000001851253ad7214f7b39e541befb6626669cb2446f`
  - **After:**     `0x000000000000000000000000d46b939123d5fb1b48ee3f90caebc9d5498ed542`
  - **Summary:**  Updates the implementation for game type 0.
  - **Detail:**  This is `gameImpls[0]` -> `0x340c1364d299ed55b193d4efcecbad8c3fb104c4`. The [`gameImpls` mapping](https://github.com/ethereum-optimism/optimism/blob/op-contracts/v3.0.0-rc.2/packages/contracts-bedrock/src/dispute/DisputeGameFactory.sol#L57) is at [storage slot 101](https://github.com/ethereum-optimism/optimism/blob/op-contracts/v3.0.0-rc.2/packages/contracts-bedrock/snapshots/storageLayout/DisputeGameFactory.json#L41) and is keyed by [`GameType` (`uint32`)](https://github.com/ethereum-optimism/optimism/blob/op-contracts/v3.0.0-rc.2/packages/contracts-bedrock/src/dispute/lib/LibUDT.sol#L224).
    - Confirm the expected key slot with the following:
      ```shell
      cast index uint32 0 101
      0xffdfc1249c027f9191656349feb0761381bb32c9f557e01f419fd08754bf5a1b
      ```
```

  ---

### [`0x16fc5058f25648194471939df75cf27a2fdc48bc`](https://github.com/ethereum-optimism/superchain-registry/blob/08e3fe429c776a532c2b6dc09571fc13e6dba5d4/superchain/configs/sepolia/op.toml#L58)  (OptimismPortal2) - Chain ID: 11155420

- **Key:**          `0x360894a13ba1a3210667c828492db98dca3e2076cc3735a920a3ca505d382bbc`
  - **Decoded Kind:**      `address`
  - **Before:** `0x2D7e764a0D9919e16983a46595CfA81fc34fa7Cd`
  - **After:** `[0xB443Da3e07052204A02d630a8933dAc05a0d6fB4](https://github.com/ethereum-optimism/superchain-registry/blob/00208555c3c356d6596feedb619da989de478ed7/validation/standard/standard-versions-sepolia.toml#L13-L13)`
  - **Summary:**           ERC-1967 implementation slot

>   - **Detail:**            Standard slot for storing the implementation address in a proxy contract that follows the ERC-1967 standard.

Reviewer: ```suggestion
  - **Detail:**            Standard slot for storing the implementation address in a proxy contract that follows the ERC-1967 standard. OptimismPortal2 contract for `op-contracts/v3.0.0-rc.2`.
```

  ---

> ### [`0x1eb2ffc903729a0f03966b917003800b145f56e2`](https://github.com/ethereum-optimism/superchain-registry/blob/b40cf4289c58e28eb1c791f9ad5724380b7516a7/validation/standard/standard-config-roles-sepolia.toml#L3)  (ProxyAdminOwner (GnosisSafe)) - Chain ID: 11155420

Reviewer: This state diff is missing the approve hash state diffs that Tenderly includes. Can you include them here? Here's an example of me doing it: https://github.com/ethereum-optimism/superchain-ops/pull/848/files#diff-ea05bc2d7951a55d5c82b5f1f3f5c02286e5a677ad7d3a7af1dc1df494ffc8edR96-R128

Reviewer: ![CleanShot 2025-04-10 at 11 17 51](https://github.com/user-attachments/assets/d03f0299-f28a-494c-99d2-1b40abc0ec51)

Reviewer: I addresses this in my latest commit.

- **Key:**          `0x0000000000000000000000000000000000000000000000000000000000000005`
  - **Decoded Kind:**      `uint256`
  - **Before:** `26`
  - **After:** `27`

>   - **Summary:**           nonce

Reviewer: ```suggestion
  - **Summary:** Nonce update
```

>   - **Detail:**

Reviewer: ```suggestion
  - **Detail:** Nonce update for the parent multisig.
```

  ---

> ### [`0x2bfb22cd534a462028771a1ca9d6240166e450c4`](https://github.com/ethereum-optimism/superchain-registry/blob/00208555c3c356d6596feedb619da989de478ed7/superchain/configs/sepolia/soneium-minato.toml#L55-L55)  (L1ERC721Bridge - Soneium Testnet Minato) - Chain ID: 1946

Reviewer: Remove as we discussed already:
```suggestion
### [`0x2bfb22cd534a462028771a1ca9d6240166e450c4`](https://github.com/ethereum-optimism/superchain-registry/blob/00208555c3c356d6596feedb619da989de478ed7/superchain/configs/sepolia/soneium-minato.toml#L55-L55)  (L1ERC721Bridge) - Chain ID: 1946
```

# Validation

This document can be used to validate the inputs and result of the execution of the upgrade transaction which you are
signing.

The steps are:

1. [Validate the Domain and Message Hashes](#expected-domain-and-message-hashes)
2. [Verifying the transaction input](#understanding-task-calldata)
3. [Verifying the state changes](#state-changes)

## Expected Domain and Message Hashes

First, we need to validate the domain and message hashes. These values should match both the values on your ledger and
the values printed to the terminal when you run the task.

> [!CAUTION]
>
> Before signing, ensure the below hashes match what is on your ledger.
>
> ### Security Council
>

> > - Domain Hash: `0xbe081970e9fc104bd1ea27e375cd21ec7bb1eec56bfe43347c3e36c5d27b8533`

Reviewer: TODO: Make sure these hashes are correct for the simulation.

> - Message Hash: `0xcedbef00e8b71f0cc711f32fe6dd303bcd5430ca36bfb66c6197ad120e790cc3`
>
> ### Optimism Foundation
>

> > - Domain Hash: `0x37e1f5dd3b92a004a23589b741196c8a214629d4ea3a690ec8e41ae45c689cbb`

Reviewer: TODO: As above.

```

> In mainnet runbooks, this calldata should appear in [Action Plan](https://x.com/gucciprayers/status/1909243252299153426) section of the Governance proposal.

Reviewer: can you remove this link and add in: https://gov.optimism.io/t/upgrade-proposal-14-isthmus-l1-contracts-mt-cannon/9796#p-43948-action-plan-9

Additionally, Safe-related nonces [will increment by one](../../../../../NESTED-VALIDATION.md#nonce-increments).

## Other Nonces
In addition to the Safe-related nonces mentioned [previously](#nested-safe-state-overrides-and-changes),
new contracts will also have a nonce value increment from 0 to 1.
This due to [EIP-161](https://eips.ethereum.org/EIPS/eip-161) which activated in 2016.

This affects the newly deployed dispute games mentioned in ["State Diffs"](#state-diffs).

## State Diffs

For each contract listed in the state diff, please verify that no contracts or state changes shown in the Tenderly diff are missing from this document. Additionally, please verify that for each contract:

- The following state changes (and none others) are made to that contract. This validates that no unexpected state
  changes occur.
- All addresses (in section headers and storage values) match the provided name, using the Etherscan and Superchain
  Registry links provided. This validates the bytecode deployed at the addresses contains the correct logic.
- All key values match the semantic meaning provided, which can be validated using the storage layout links provided.

> <pre>

Reviewer: You should delete everything below this line (`<pre>` tags).

  <code>

> ----- DecodedStateDiff[0] -----

Reviewer: Note, when comparing the state diffs with tenderly, you may notice state diffs in Tenderly that don't appear in the auto generated state diffs. We must add them in with an explanation of what they are. 

e.g. search for `TENDERLY ONLY STATE DIFF` in this file to see an example: https://github.com/ethereum-optimism/superchain-ops/blob/main/src/improvements/tasks/eth/001-opcm-upgrade-v200/VALIDATION.md

  - **Summary:**           ERC-1967 implementation slot
  - **Detail:**            Standard slot for storing the implementation address in a proxy contract that follows the ERC-1967 standard. SystemConfig contract for `op-contracts/v3.0.0-rc.2`.

  ---

### [`0x05c993e60179f28bf649a2bb5b00b5f4283bd525`](https://github.com/ethereum-optimism/superchain-registry/blob/00208555c3c356d6596feedb619da989de478ed7/superchain/configs/sepolia/ink.toml#L59-L59) (SystemConfig) - Chain ID: 763373

- **Key:**          `0x360894a13ba1a3210667c828492db98dca3e2076cc3735a920a3ca505d382bbc`
  - **Decoded Kind:**      `address`
  - **Before:** `0x760C48C62A85045A6B69f07F4a9f22868659CbCc`
  - **After:** `[0x340f923E5c7cbB2171146f64169EC9d5a9FfE647](https://github.com/ethereum-optimism/superchain-registry/blob/00208555c3c356d6596feedb619da989de478ed7/validation/standard/standard-versions-sepolia.toml#L9-L9)`
  - **Summary:**           ERC-1967 implementation slot
  - **Detail:**            Standard slot for storing the implementation address in a proxy contract that follows the ERC-1967 standard. SystemConfig contract for `op-contracts/v3.0.0-rc.2`.

> 

Reviewer 2: current format breaks hyperlinking

```suggestion
  - **After:** [`0x340f923E5c7cbB2171146f64169EC9d5a9FfE647`](https://github.com/ethereum-optimism/superchain-registry/blob/00208555c3c356d6596feedb619da989de478ed7/validation/standard/standard-versions-sepolia.toml#L9-L9)
```

```

In mainnet runbooks, this calldata should appear in [Action Plan](https://gov.optimism.io/t/upgrade-proposal-14-isthmus-l1-contracts-mt-cannon/9796#p-43948-action-plan-9) section of the Governance proposal.

# State Changes

## Nested Safe State Overrides and Changes

> <!--TODO(#740) - I see additional overrides to the ProxyAdminSafe not documented in NESTED-VALIDATION.md -->

Author: Was an old to-do carried over, don't believe its needed now

Additionally, Safe-related nonces [will increment by one](../../../../../NESTED-VALIDATION.md#nonce-increments).

## Other Nonces
In addition to the Safe-related nonces mentioned [previously](#nested-safe-state-overrides-and-changes),
new contracts will also have a nonce value increment from 0 to 1.
This due to [EIP-161](https://eips.ethereum.org/EIPS/eip-161) which activated in 2016.

This affects the newly deployed dispute games mentioned in ["State Diffs"](#state-diffs).

## State Diffs

For each contract listed in the state diff, please verify that no contracts or state changes shown in the Tenderly diff are missing from this document. Additionally, please verify that for each contract:

- The following state changes (and none others) are made to that contract. This validates that no unexpected state
  changes occur.
- All addresses (in section headers and storage values) match the provided name, using the Etherscan and Superchain
  Registry links provided. This validates the bytecode deployed at the addresses contains the correct logic.
- All key values match the semantic meaning provided, which can be validated using the storage layout links provided.

  ---
  

> ### `[0x034edd2a225f7f429a63e0f1d2084b9e0a93b538](https://github.com/ethereum-optimism/superchain-registry/blob/1a5d7a208cea9b0ea175df1fe71bdc4da7f4c04c/superchain/configs/sepolia/op.toml#L5) (SystemConfig - OP Sepolia Testnet)`

Author: Fixed

  
- **Summary:**           ERC-1967 implementation slot
  - **Detail:**            Standard slot for storing the implementation address in a proxy contract that follows the ERC-1967 standard.
  
  ---
  
### `[0x05c993e60179f28bf649a2bb5b00b5f4283bd525](https://github.com/ethereum-optimism/superchain-registry/blob/00208555c3c356d6596feedb619da989de478ed7/superchain/configs/sepolia/ink.toml#L59-L59) (SystemConfig - Ink Sepolia Testnet)`
  
#### Decoded State Change: 1
  - **Contract:**          `SystemConfig - Ink Sepolia Testnet`
  - **Chain ID:**          `763373`
  
- **Key:**          `0x360894a13ba1a3210667c828492db98dca3e2076cc3735a920a3ca505d382bbc`
  - **Decoded Kind:**      `address`
  - **Before:** `0x760C48C62A85045A6B69f07F4a9f22868659CbCc`
  - **After:** `[0x340f923E5c7cbB2171146f64169EC9d5a9FfE647](https://github.com/ethereum-optimism/superchain-registry/blob/00208555c3c356d6596feedb619da989de478ed7/validation/standard/standard-versions-sepolia.toml#L9-L9)`
  
- **Summary:**           ERC-1967 implementation slot
  - **Detail:**            Standard slot for storing the implementation address in a proxy contract that follows the ERC-1967 standard.

  ---
  

> ### `[0x05f9613adb30026ffd634f38e5c4dfd30a197fa1](https://github.com/ethereum-optimism/superchain-registry/blob/00208555c3c356d6596feedb619da989de478ed7/superchain/configs/sepolia/op.toml#L63-L63) (DisputeGameFactory)`

Reviewer: Are you specifically talking about the `#### Decoded State Change:` headings? This time around I've grouped state changes by address. This is what it looks like in my pr and what it should also look like in this one. 
![CleanShot 2025-04-09 at 09 38 35](https://github.com/user-attachments/assets/4d722f61-d0cf-44ba-8eb5-36c071ebed2f)

> ### `[0x05f9613adb30026ffd634f38e5c4dfd30a197fa1](https://github.com/ethereum-optimism/superchain-registry/blob/00208555c3c356d6596feedb619da989de478ed7/superchain/configs/sepolia/op.toml#L63-L63) (DisputeGameFactory)`

Reviewer 2: I'm a bit confused by the screenshot tbh. It seems to be grouping diffs in different contracts together. 

The high level structure I'm proposing is (the numbers aren't important here, just the nesting structure. 

- Contract 1
  - diff 1
  - diff 2
- Contract 2 
  - diff 1
- Contract 3
  - diff 1
  - diff 2
  - diff 3

> ### `[0x05f9613adb30026ffd634f38e5c4dfd30a197fa1](https://github.com/ethereum-optimism/superchain-registry/blob/00208555c3c356d6596feedb619da989de478ed7/superchain/configs/sepolia/op.toml#L63-L63) (DisputeGameFactory)`

Reviewer: Okay yea that makes sense. Can we update the markdown file with this as a fix too so other task devs pick up these changes? I can work on getting that in.

> ### `[0x05f9613adb30026ffd634f38e5c4dfd30a197fa1](https://github.com/ethereum-optimism/superchain-registry/blob/00208555c3c356d6596feedb619da989de478ed7/superchain/configs/sepolia/op.toml#L63-L63) (DisputeGameFactory)`

Reviewer 2: Yeah, agree we should. Ideally we just update how these are printed if possible.

> ### `[0x05f9613adb30026ffd634f38e5c4dfd30a197fa1](https://github.com/ethereum-optimism/superchain-registry/blob/00208555c3c356d6596feedb619da989de478ed7/superchain/configs/sepolia/op.toml#L63-L63) (DisputeGameFactory)`

Author: Fixed

  
- **Summary:**           ERC-1967 implementation slot
  - **Detail:**            Standard slot for storing the implementation address in a proxy contract that follows the ERC-1967 standard.
  
  ---
  
### `[0x3454f9df5e750f1383e58c1cb001401e7a4f3197](https://github.com/ethereum-optimism/superchain-registry/blob/00208555c3c356d6596feedb619da989de478ed7/superchain/configs/sepolia/ink.toml#L53-L53) (AddressManager - Ink Sepolia)`
  
#### Decoded State Change: 8
  - **Contract:**          `AddressManager`
  - **Chain ID:**          `763373`
  
- **Key:**          `0x515216935740e67dfdda5cf8e248ea32b3277787818ab59153061ac875c9385e`
  - **Before:**     `0x0000000000000000000000003ea6084748ed1b2a9b5d4426181f1ad8c93f6231`
  - **After:**     `0x0000000000000000000000005d5a095665886119693f0b41d8dfee78da033e8b`
  
- **Summary:**  The name `OVM_L1CrossDomainMessenger` is set to the address of the new 'op-contracts/v3.0.0-rc.2' L1CrossDomainMessenger at [0x5d5a095665886119693f0b41d8dfee78da033e8b](https://github.com/ethereum-optimism/superchain-registry/blob/1a5d7a208cea9b0ea175df1fe71bdc4da7f4c04c/validation/standard/standard-versions-sepolia.toml#L18).

>   - **Detail:** This key is complicated to compute, so instead we attest to correctness of the key by
>                      verifying that the "Before" value currently exists in that slot, as explained below.
>                      **Before** address matches both of the following cast calls:
>                       1. What is returned by calling `AddressManager.getAddress()`:
>                        - cast call 0x3454F9df5E750F1383e58c1CB001401e7A4f3197 'getAddress(string)(address)' 'OVM_L1CrossDomainMessenger' --rpc-url sepolia
>                       2. What is currently stored at the key:
>                        - cast storage 0x3454F9df5E750F1383e58c1CB001401e7A4f3197 0x515216935740e67dfdda5cf8e248ea32b3277787818ab59153061ac875c9385e --rpc-url sepolia

Reviewer 2: @blmalone WDYT of this change?

```suggestion
  - **Detail:** **THIS WAS CAREFULLY VERIFIED BY RUNBOOK REVIEWERS AND NEED NOT BE CHECKED BY SIGNERS.**
    This key is complicated to compute, so instead we attest to correctness of the key by
    verifying that the "Before" value currently exists in that slot, as explained below.
    **Before** address matches the following cast call to `AddressManager.getAddress()`:
      - `cast call 0x3454F9df5E750F1383e58c1CB001401e7A4f3197 'getAddress(string)(address)' 'OVM_L1CrossDomainMessenger' --rpc-url sepolia`
      - returns: `0x3eA6084748ED1b2A9B5D4426181F1ad8C93F6231`
```

Reviewer: Yes, I'm all for giving signers explicit messages so that it makes their lives easier. Lets add this in.
